### PR TITLE
Edge case gates for IEEET1 (Validation)

### PR DESCRIPTION
**Pull request by lukelowry**
## Description
 
A few edge-case guards for parameters that caused the unsimplified versions of our examples to fail.
 
## Proposed changes

- Apply a minimum time constant to `Tr`, `Ta`, `Te`, and `Tf` to prevent singular residual equations.
- Treat `Se1 = Se2 = 0` as disabled saturation and validate enabled saturation parameters before deriving the coefficients.
- Document the parameter handling and add a test for zero time constants and disabled saturation.

## Checklist

- [x] All tests pass for the focused PhasorDynamics coverage.
- [x] Code compiles cleanly with flags `-Wall -Wpedantic -Wconversion -Wextra`.
- [x] The new code follows GridKit™ style guidelines.
- [x] There are unit tests for the new code.
- [x] The new code is documented.
- [x] The feature branch is rebased with respect to the target branch.

Changelog changes N/A; this is a minor correction to an existing model.

## Further comments

This preserves the existing fixed-size IEEET1 structure and allows the unsimplified example data to run without case-file changes. A structural treatment of zero time constants can be considered separately and is the preferred approach. I noticed that making the time constant nonzero, rather than zero, dramatically slows the simulation.

### CODE REVIEW

File: GridKit/Model/PhasorDynamics/Exciter/IEEET1/README.md [hunk 1]
```diff
@@ -147,7 +175,14 @@ Here $q$ is GridKit's [Quadratic Ramp](../../../../CommonMath.md#primitives).
 
 ## Initialization
 
-The implementation first applies $T_R \leftarrow \max(T_R, 10^{-3})$. The machine initializes $E_{fd}$ first. IEEET1 reads that value as $E_{fd,0}$, along with any attached $\omega$ and $V_S$, and solves the steady-state algebraic chain so all residuals vanish with $\dot y = 0$. The sensed terminal voltage initializes from the positive bus-voltage magnitude. Saturation and the speed-limit flag are included directly; $V_\text{ref}$ is set to close the $V_{tr}$ equation with the current input values.
+The implementation first applies $T \leftarrow \max(T, 10^{-3})$ for
+$T \in \{T_R, T_A, T_E, T_F\}$. This should be replaced with a structural tempalte change in the future. The machine initializes $E_{fd}$ first. IEEET1
+reads that value as $E_{fd,0}$, along with any attached $\omega$ and $V_S$, and
```
Comment (pelesh) [inline GridKit/Model/PhasorDynamics/Exciter/IEEET1/README.md]:
```suggestion
$T \in \{T_R, T_A, T_E, T_F\}$. This should be replaced with a structural template change in the future.
The machine initializes $E_{fd}$ first. IEEET1
reads that value as $E_{fd,0}$, along with any attached $\omega$ and $V_S$, and
```

Comment (pelesh) [reply GridKit/Model/PhasorDynamics/Exciter/IEEET1/README.md]: tempalte -> template

**Review by pelesh — APPROVED** [review]

**Comment by lukelowry** [comment]
@pelesh Fixed and rebased, thank you.
